scipy/optimize/_qap.py

```diff
@@ -0,0 +1,418 @@
+    to a sign change) [1]_.
+
+    This algorithm can be thought of as finding an alignment of the
+    vertices of two graphs which minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements. The option to add seeds
+    (known vertex correspondence between some nodes) is also available
+    [2]_.
+
+    Note that the quadratic assignment problem is NP-hard, is not
+    known to be solvable in polynomial time, and is computationally
+    intractable. Therefore, the results given are approximations,
+    not guaranteed to be exact solutions.
+
+
+    Parameters
+    ----------
+    cost_matrix : 2d-array, square, non-negative
+        A square adjacency matrix
+
+    dist_matrix : 2d-array, square, non-negative
+        A square adjacency matrix
+
+    seed : 2d-array, optional, (default = None)
+        Allows the user apply a seed, fixing part of the matching between
+        the two adjacency matrices.
+        For column 1, each entry is an index of a node in `cost_matrix`.
+        For column 2, each entry is an index of a node in `dist_matrix`.
+        The elements of ``seed[:, 0]`` and ``seed[:, 1]`` are vertices
+        which are known to be matched, that is, `seed[i, 0]` is matched to
+        vertex `seed[i, 1]`. Array shape (m , 2) where m <= number of nodes
+
+    maximize : bool (default = False)
+        Gives users the option to solve the Graph Matching Problem (GMP)
+        rather than QAP. This is accomplished through trivial negation
+        of the objective function.
+
+
+    options : dict, optional
+        A dictionary of solver options. All methods accept the following
+        options:
+
+            n_init : int, positive (default = 1)
+                Number of random initializations of the starting
+                permutation matrix that the FAQ algorithm will undergo.
+            init_method : string (default = 'barycenter')
+                The algorithm may be sensitive to the initial permutation
+                matrix (or search position) chosen due to the possibility
+                of several local minima within the feasible region.
+                With only 1 initialization, a barycenter init will
+                likely return a more accurate permutation.
+                Choosing several random initializations as opposed to
+                the non-informative barycenter will likely result in a
+                more accurate result at the cost of higher runtime.
+                The initial position chosen:
+                "barycenter" : the non-informative "flat doubly stochastic
+                matrix,":math:`J=1*1^T /n` , i.e the barycenter of the
+                feasible region (where n is the number of nodes and '1' is
+                a (n, 1) array of ones)
+                "rand" : some random point near :math:`J, (J+K)/2`, where K
+                is some random doubly stochastic matrix
```

Comment:
I will likely have more comments about the description of this, but I will save them for when we figure out what is going to go here. The guiding comment will be that this needs to be understandable. What is this "position"?  If this is like other iterative algorithms, when I think of initialization, I'm thinking of the initial values of the decision variables, which will be iteratively refined in the optimization process. In this case, this might be an array of column indices like the output `col_ind`. Is that right? What is the relationship between J and these initial values? By "some random point near J, (J+K)/2"... do you mean that the random point is defined as (J+K)/2, or do you mean that the coordinates of this point are (J, (J+K)/2) (whatever _that_ means... I'm not familiar with using matrices as coordinates...)?

Reply:
Here the "position" refers to the `P` matrix in the objective function. The algorithm iteratively adjusts the initialized matrix through the FW loop, minimizing the objective function value. After convergence, `P`, which is a doubly stochastic matrix at this point, is projected onto the set of permutation matrices, from which the output `col_ind` is determined.

Also, the random point is defined as (J+K)/2. It was a bit confusing the documentation, so I've adjusted it.